### ISSUE TITLE
play.api.mvc.Session uses global app state

### ISSUE THREAD

**Issue by lobax** [playframework/playframework#9816]
### Play Version
2.7.3

### API
Scala

### Operating System
SMP Debian 5.2.17-1kali2 (2019-10-17) x86_64 GNU/Linux

### JDK
openjdk 11.0.5 2019-10-15

### Expected Behavior
Session methods should be either deprecated or not use plays global state. 

I.e. methods like `Session.decode` should be usable without having to set `play.allowGlobalApplication = true`

### Actual Behavior
A runtime exception is thrown:

```
java.lang.RuntimeException:
The global application reference is disabled. Play's global state is deprecated and will
be removed in a future release. You should use dependency injection instead. To enable
the global application anyway, set play.allowGlobalApplication = true.

at scala.sys.package$.error(package.scala:30)
at play.api.Play$.privateMaybeApplication(Play.scala:90)
at play.api.libs.Crypto$.cookieSigner(Crypto.scala:21)
at play.api.mvc.Session$.signedCodec$lzycompute(Session.scala:162)
at play.api.mvc.Session$.signedCodec(Session.scala:162)
at play.api.mvc.Session$.signedCodec(Session.scala:149)
at play.api.mvc.FallbackCookieDataCodec.decode(Cookie.scala:803)
at play.api.mvc.FallbackCookieDataCodec.decode$(Cookie.scala:790)
at play.api.mvc.Session$.decode(Session.scala:149)
```

### Reproducible Test Case

Use `Session.decode()`, `Session.encode()` etc as described [here](https://www.playframework.com/documentation/2.7.x/api/scala/play/api/mvc/Session$.html#encode(data:Map[String,String]):String)

**Comment by renatocaval** [comment]
Thanks for reporting this. 

Indeed, that should not be the case. We will need to investigate this further and see how we can remove the dependency on the global application.

**Comment by renatocaval** [comment]
hi @lobax, 

I look further to understand why you are not getting a deprecation warning when calling `Session.decode`. As you can see [here](https://github.com/playframework/playframework/blob/2.7.x/core/play/src/main/scala/play/api/mvc/Session.scala#L149-L193), all methods in `Session` are deprecated and we recommend to inject `SessionCookieBaker` instead. 

However, the `decode` method is part of `CookieDataCodec` which is not deprecated. 
The problem is that in order to implement `CookieDataCodec`, the `Session` object relies on the global state and then it blows up at runtime. 

`SessionCookieBaker` also extends `CookieDataCodec`. So instead, you should inject the `SessionCookieBaker`. 

In Play 2.8, the deprecated methods are removed and you won't even have a `Session.decode`. So I suggest to inject `SessionCookieBaker` and set `play.allowGlobalApplication` back to `false`. 

I will see if I can add something on Play 2.7 to make it more obvious that those calls are deprecated.

**Comment by mkurz** [comment]
Fixed in #9865 long time ago.
